src/read_user_input_validate.c

```diff
@@ -0,0 +1,77 @@
+#include "../include/read_user_input_validate.h"
+#include "../include/globals.h"
+#include "../include/structures.h"
+#include <stdlib.h>
+#include <stdio.h>
+//#include <time.h>
+
+
+/**
+ * @file read_user_input_validate.c
+ *
+ * @author Ritika Arora
+ *
+ * @brief Reads the input from the console and adds to the passenger struct after validating.
+ *
+ * @details Reads the passenger details from the user console. Validates the inputs given.
+ * Then, adds the inputs taken to the passenger structure
+ *
+ * @param[in] passenger_info pointer to the passenger structure containing data of the passenger.
+ * @param[in] timer to get the time of simulation and store as time of arrival
+ *
+ * @return void returns nothing
+*/
+void read_user_input_validate(struct passenger *passenger_info, unsigned int timer){
+
+
+    int arrival_floor, dest_floor, passenger_id; // local variables to get the values from user and store them in the structure
+
+
+    printf("Enter the id");
+
+    while((scanf("%d",&passenger_id))==0){
+    	//To check that user enters an integer value only
+    		scanf("%*[^\n]");
+    		printf("Invalid entry, please enter a valid number: ");
+    }
+    if(passenger_id <=0){
+        	printf("Invalid passenger id entered");
+        	exit(1);
+    }
+    printf("Enter the arrival floor between 1 and %d:",elevator_count);
+
+    while((scanf("%d",&arrival_floor))==0){
+    	//To check that user enters an integer value only
+        	scanf("%*[^\n]");
+        	printf("Invalid entry, please enter a valid number: ");
+    }
+    if(arrival_floor<=0 || arrival_floor > elevator_count){
+    		printf("Invalid arrival floor entered!!");
+    		exit(1);
+    }
+    printf("Enter the destination floor between 1 and %d:",elevator_count);
+
+    while((scanf("%d",&dest_floor))==0){
+    	//To check that user enters an integer value only
+        	scanf("%*[^\n]");
+        	printf("Invalid entry, please enter a valid number: ");
+    }
+    if(dest_floor <=0 || dest_floor > elevator_count){
```

Comment:
if(dest_floor <=0 || dest_floor > floor_count)

Reply:
> global variable elevator_count needs to be replaced with global variable floor_count

Modified the file